polar-core/src/vm.rs

```diff
@@ -758,37 +795,46 @@ impl PolarVirtualMachine {
     }
 
     fn log(&self, message: &str, terms: &[&Term]) {
-        self.log_with(|| message, terms)
+        self.log_with(LogLevel::Info, || message, terms)
     }
 
-    fn log_with<F, R>(&self, message_fn: F, terms: &[&Term])
+    fn log_with<F, R>(&self, level: LogLevel, message_fn: F, terms: &[&Term])
     where
         F: FnOnce() -> R,
         R: AsRef<str>,
     {
-        if self.polar_log && !self.polar_log_mute {
-            let mut indent = String::new();
-            for _ in 0..=self.queries.len() {
-                indent.push_str("  ");
-            }
-            let message = message_fn();
-            let lines = message.as_ref().split('\n').collect::<Vec<&str>>();
-            if let Some(line) = lines.first() {
-                let mut msg = format!("[debug] {}{}", &indent, line);
-                if !terms.is_empty() {
-                    let relevant_bindings = self.relevant_bindings(terms);
-                    msg.push_str(&format!(
-                        ", BINDINGS: {{{}}}",
-                        relevant_bindings
-                            .iter()
-                            .map(|(var, val)| format!("{} => {}", var.0, val.to_polar()))
-                            .collect::<Vec<String>>()
-                            .join(", ")
-                    ));
+        if let Some(configured_log_level) = self.log_level {
+            // preserve the old `polar_log_mute` behavior which omits parameter
+            // specialization checking Unify, IsA and other events from the log
+            if level == LogLevel::Trace && self.polar_trace_mute {
+                return;
+            }
+            if configured_log_level.should_print_on_level(level) {
+                let mut indent = String::new();
+                for _ in 0..=self.queries.len() {
+                    indent.push_str("  ");
                 }
-                self.print(msg);
-                for line in &lines[1..] {
-                    self.print(format!("[debug] {}{}", &indent, line));
+                let message = message_fn();
+                let lines = message.as_ref().split('\n').collect::<Vec<&str>>();
+                if let Some(line) = lines.first() {
+                    let mut msg = format!("[{}] {}{}", level, &indent, line);
```

Comment:
I wonder if we should identify our logs somehow, like `[oso - debug] ...`. Thinking about the case where you're running your web server in development mode w/ `POLAR_LOG=info` and Oso's logs are getting interspersed with the server's logs.

Reply:
great call! will add a prefix here